src/zkevm_specs/util/ec.py

```diff
@@ -0,0 +1,113 @@
+from typing import Tuple
+from .arithmetic import FQ
+from eth_keys import KeyAPI  # type: ignore
+
+
+class WrongFieldInteger:
+    """
+    Wrong Field arithmetic Integer, representing the implementation at
+    https://github.com/privacy-scaling-explorations/halo2wrong/blob/master/integer/src/integer.rs
+    """
+
+    limbs: Tuple[FQ, FQ, FQ, FQ]  # Little-Endian limbs of [72, 72, 72, 40] bits
+    le_bytes: bytes  # Little-Endian bytes
+
+    def __init__(self, value: int) -> None:
+        mask = (1 << 72) - 1
+        l0 = (value >> 0 * 72) & mask
+        l1 = (value >> 1 * 72) & mask
+        l2 = (value >> 2 * 72) & mask
+        l3 = (value >> 3 * 72) & mask
+        self.limbs = (FQ(l0), FQ(l1), FQ(l2), FQ(l3))
+        self.le_bytes = value.to_bytes(32, "little")
+
+    def to_le_bytes(self) -> bytes:
+        (l0, l1, l2, l3) = self.limbs
+        val = l0.n + (l1.n << 1 * 72) + (l2.n << 2 * 72) + (l3.n << 3 * 72)
+        return val.to_bytes(32, "little")
+
+    def to_be_bytes(self) -> bytes:
+        (l0, l1, l2, l3) = self.limbs
+        val = l0.n + (l1.n << 1 * 72) + (l2.n << 2 * 72) + (l3.n << 3 * 72)
+        return val.to_bytes(32, "big")
+
+
+class Secp256k1BaseField(WrongFieldInteger):
+    """
+    Secp256k1 Base Field.
+    """
+
+    def __init__(self, value: int) -> None:
+        WrongFieldInteger.__init__(self, value)
+
+
+class Secp256k1ScalarField(WrongFieldInteger):
+    """
+    Secp256k1 Scalar Field.
+    """
+
+    def __init__(self, value: int) -> None:
+        WrongFieldInteger.__init__(self, value)
+
+
+# TODO: There is another one used in tx_circuit, try to merge into one.
+#       Reminder: endianness of public key is differ with the one in tx_circuit
+class ECDSAVerifyChip:
```

Comment:
This was copied from tx_circuit and only change the endianness of public key ([here](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/495/files#diff-1ea18c63b94bf02b9fb4a015eb57ad3f17fc830702a79a5e9aaad4e908971208R82-R83)  and [here](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/495/files#diff-1ea18c63b94bf02b9fb4a015eb57ad3f17fc830702a79a5e9aaad4e908971208R82-R83))